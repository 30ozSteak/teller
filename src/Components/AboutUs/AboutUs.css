.about-slider {
  height: 650px;
  margin: -2rem auto 0;
  position: relative;
  overflow: hidden;
  width: 375px;
  white-space: nowrap;
}

.about-slider-wrapper {
  height: 100%;
  position: relative;
  width: 100%;
}

.about-arrow-box {
  align-items: flex-end;
  bottom: 1rem;
  display: flex;
  justify-content: space-between;
  left: 50%;
  position: fixed;
  text-align: center;
  transform: translate(-50%, -50%);
  width: 15rem;
  z-index: 10;
}

.right-caret,
.left-caret {
  height: 2rem;
  width: 1.7rem;
}

.caret {
  cursor: pointer;
  transition: 0.2s;
  height: 2rem;
  width: 2rem;
}

.caret:hover {
  filter: brightness(1.5);
}

.left-c {
  transform: rotate(-90deg);
}
.right-c {
  transform: rotate(90deg);
}
