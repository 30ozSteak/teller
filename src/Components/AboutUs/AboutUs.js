import React, { Component } from "react";
import { LeftCaret, RightCaret } from "../Onboarding/OnboardingArrows";
import caret from "../../Assets/caret.svg";
import dev1 from "../../Assets/images/autumn.png";
import dev3 from "../../Assets/images/dina.png";
import dev2 from "../../Assets/images/daniel.png";
import dev4 from "../../Assets/images/nick.png";
import About from "../About/About";
import "./AboutUs.css";

class AboutUs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      images: [dev1, dev2, dev3, dev4],
      currentIndex: 0,
      translateValue: 0
    };
  }

  prevSlide = () => {
    if (!this.state.currentIndex) {
      return this.state({
        currentIndex: 0,
        translateValue: 0
      });
    }

    this.setState(prevState => ({
      currentIndex: prevState.currentIndex - 1,
      translateValue: prevState.translateValue + this.slideWidth()
    }));
  };
  nextSlide = () => {
    if (this.state.currentIndex === this.state.images.length - 1) {
      return this.setState({
        currentIndex: 0,
        translateValue: 0
      });
    }
    this.setState(prevState => ({
      currentIndex: prevState.currentIndex + 1,
      translateValue: prevState.translateValue + -this.slideWidth()
    }));
  };

  slideWidth = () => {
    return document.querySelector("./aboutSlide").clientWidth;
  };

  render() {
    const aboutSlider = this.state.images.map((image, i) => {
      return <About key={i} image={image} />;
    });

    return (
      <div className="about-slider">
        <div
          className="about-slider-wrapper"
          style={{
            transform: `translateX(${this.state.translateValue}px)`,
            transition: "transform ease-out 0.5s"
          }}
        >
          {aboutSlider}
        </div>
        <div className="about-arrow-box">
          <LeftCaret prevSlide={this.prevSlide} />

          <RightCaret nextSlide={this.nextSlide} />
        </div>
      </div>
    );
  }
}

{
}
export default AboutUs;
