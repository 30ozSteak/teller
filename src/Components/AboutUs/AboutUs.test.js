import React from 'react'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import AboutUs from './AboutUs'
import { shallow } from 'enzyme'

describe('AboutUs Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<AboutUs />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})