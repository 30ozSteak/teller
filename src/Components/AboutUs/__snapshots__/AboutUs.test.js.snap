// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AboutUs />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Meet the Team
        </h1>,
        <h2>
          Dina Caraballo
        </h2>,
        <h2>
          Nick D'Ambrosio
        </h2>,
        <h2>
          Autumn Martin
        </h2>,
        <h2>
          Daniel Mulitauopele
        </h2>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Meet the Team",
        },
        "ref": null,
        "rendered": "Meet the Team",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Dina Caraballo",
        },
        "ref": null,
        "rendered": "Dina Caraballo",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Nick D'Ambrosio",
        },
        "ref": null,
        "rendered": "Nick D'Ambrosio",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Autumn Martin",
        },
        "ref": null,
        "rendered": "Autumn Martin",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Daniel Mulitauopele",
        },
        "ref": null,
        "rendered": "Daniel Mulitauopele",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Meet the Team
          </h1>,
          <h2>
            Dina Caraballo
          </h2>,
          <h2>
            Nick D'Ambrosio
          </h2>,
          <h2>
            Autumn Martin
          </h2>,
          <h2>
            Daniel Mulitauopele
          </h2>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Meet the Team",
          },
          "ref": null,
          "rendered": "Meet the Team",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dina Caraballo",
          },
          "ref": null,
          "rendered": "Dina Caraballo",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Nick D'Ambrosio",
          },
          "ref": null,
          "rendered": "Nick D'Ambrosio",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Autumn Martin",
          },
          "ref": null,
          "rendered": "Autumn Martin",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Daniel Mulitauopele",
          },
          "ref": null,
          "rendered": "Daniel Mulitauopele",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
